x-pack/plugins/cloud_security_posture/public/pages/findings/findings.tsx

```diff
@@ -22,6 +24,10 @@ export const FindingsNoPageTemplate = () => {
   const dataViewQuery = useLatestFindingsDataView();
   // TODO: Consider splitting the PIT window so that each "group by" view has its own PIT
   const { pitQuery, pitIdRef, setPitId } = useFindingsEsPit('findings');
+  const getSetupStatus = useCspSetupStatusApi();
+
+  const hasFindings = getSetupStatus.data?.status === 'indexed';
+  if (!hasFindings) return <NoFindingsStates />;
```

Comment:
This logic will exist in every page using the `NoFindingStates` component, consider adding it into the `NoFindingStates` component itself, and passing `children` into it to render when the status is `indexed` (logically similar to how `CloudPosturePage` works)

Reply:
Wanted to do it at first but it's starting to make the CloudPosturePage `query` property a bit messy.. passing 3 queries that depend on each other in the findings page is a bit much. I think for now this keeps pages relatively clean. that's the main reason.

a second and less important reason to keep it like that is that in the future we will probably want some kind of banner telling the user that it needs to reinstall the package, or redeploy agents (in cases where we get `indexed` but the package was uninstalled/ agents are not working). checking this hook will be necessary.